### PR TITLE
Kops 1.24.1

### PR DESCRIPTION
**Pull request by mikesplain**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Fixes https://github.com/Homebrew/homebrew-core/pull/105381

### CODE REVIEW

File: Formula/kops.rb [hunk 1]
```diff
@@ -25,11 +25,11 @@ class Kops < Formula
 
   def install
     ENV["VERSION"] = version unless build.head?
-    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = buildpath
     kopspath = buildpath/"src/k8s.io/kops"
     kopspath.install Dir["*"]
     system "make", "-C", kopspath
-    bin.install "bin/kops"
+    bin.install "kops"
```
Comment (SMillerDev) [inline Formula/kops.rb:32]: What does this fix exactly?

Comment (mikesplain) [reply Formula/kops.rb:32]: As far as I can tell, kops now builds based on `$GOBIN` path which is why `GOBIN` was subbed for `GOPATH`. That change caused kops to build into the `buildpath` directory which no longer includes being in a `/bin` folder during the build process.  

See: https://github.com/kubernetes/kops/commit/5b90e580d099b66d4742df3dbb2e220e7b514a8a#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R99 for the main change that I think affected this.

I'm open to suggestions if you know better solutions!

**Comment by mikesplain** [comment]
I'm going to push a fix with the right commit message. A release occurred while I was opening this.

File: Formula/kops.rb (hunk 1 above)
Comment (carlocab) [reply Formula/kops.rb:32]: Their `Makefile` just seems to call `go build`. Can we just do that directly and align this formulae with the many other formulae that build using `go`?

Comment (mikesplain) [reply Formula/kops.rb]: Fair enough, I believe we couldn't historically for some reason but I've modified the PR to a simple go build. Let me know if you if you have any follow ups to those changes, thanks!

**Review by carlocab — APPROVED** [review]
Nice work. Thanks @mikesplain!

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**